File: Objective-C/Reference App/303. Get turn-by-turn navigation route data/GetRouteViewController.m
//
//  GetRouteViewController.m
//  Reference App
//
//  Created by David Deller on 5/3/18.
//  Copyright © 2018 TallyGo. All rights reserved.
//

#import "GetRouteViewController.h"

@interface GetRouteViewController ()

@end

@implementation GetRouteViewController

- (IBAction)go:(id)sender {
    [self getRoute];
}

- (void)getRoute {
    // Get these coordinates from your app, these are just a sample
    TGWaypoint *origin = [TGWaypoint.alloc initWithCoordinate:CLLocationCoordinate2DMake(34.101558, -118.340944)]; // Grauman's Chinese Theatre
    TGWaypoint *destination = [TGWaypoint.alloc initWithCoordinate:CLLocationCoordinate2DMake(34.011441, -118.494932)]; // Santa Monica Pier
    
    TGRouteRequest *request = [TGRouteRequest.alloc initWithOriginWaypoint:origin destinationWaypoint:destination];
    
    [TGNavigationService routeForRequest:request completionHandler:^(TGRouteResponse * _Nonnull response) {
        if (response.error != nil) {
            // handle error
            
            [self handleError:response.error];
            
        } else {
            TGRoute *route = response.route;
            // do something with route
            
            [self handleOutput:route];
        }
    }];

}

@end

File: Swift 4/Reference App/303. Get turn-by-turn navigation route data/GetRouteViewController.swift
//
//  GetRouteViewController.swift
//  Reference App
//
//  Created by David Deller on 5/1/18.
//  Copyright © 2018 TallyGo. All rights reserved.
//

import UIKit
import TallyGoKit

class GetRouteViewController: OutputViewController {
    
    @IBAction func go(_ sender: Any) {
        getRoute()
    }
    
    // MARK: -

    func getRoute() {
        // Get these coordinates from your app, these are just a sample
        let origin = TGWaypoint(coordinate: CLLocationCoordinate2D(latitude: 34.101558, longitude: -118.340944)) // Grauman's Chinese Theatre
        let destination = TGWaypoint(coordinate: CLLocationCoordinate2D(latitude: 34.011441, longitude: -118.494932)) // Santa Monica Pier
        
        let request = TGRouteRequest(originWaypoint: origin, destinationWaypoint: destination)
        
        TGNavigationService.route(for: request) { (response) in
            if let error = response.error {
                // handle error
                
                self.handleError(error)
                
            } else if let route = response.route {
                // do something with route
                
                self.handleOutput(route)
            }
        }
    }

}
